dspinner/pin_test.go

```diff
@@ -0,0 +1,1056 @@
+package dspinner
+
```

Comment:
The intention was not to invest too much time into de-duplicating the test logic, since I was thinking that the ipldpinner world go away after some time.  If we want to keep both, or may have others the future, then common test logic should be moved to common location.

Will do in subsequent PR.

Reply:
SGTM